Migration and countersigning content updates, 11ty site, take 2 (#357)
holochain/docs-pages: src/pages/references/glossary.md

#### Client

In Holochain terms, any piece of external software that accesses a [DNA's](#dna) [zome functions](#zome-function). The client makes function calls over the [conductor's](#conductor) [RPC interface](#rpc-interface), and can be a [GUI](#graphical-user-interface-gui), shell script, service, or scheduled task. This client needs to be granted a [capability](#capability) to make these calls, because the RPC interface represents the user's [agency](#agency) in the application, and access to it should be restricted. For this reason, a client also usually lives on the same machine as the [conductor](#conductor).

#### Client/server

A highly [centralized](#centralization) [distributed system](#distributed-system) architecture in which certain privileged nodes are responsible for most of the processing, storage, and decision-making. Client/server systems typically give low [agency](#agency) to end-users, due to the fact that the privileged nodes mediate most user interactions.

#### Cloning

> The act of modifying one or more [DNA modifiers](#dna-modifiers) at DNA installation time in order to change its [hash](#dna-hash) and optionally its behavior via [DNA properties](#dna-properties), thus creating an isolated [network](#network) and [DHT](#distributed-hash-table-dht) for that clone. This allows all [agents](#agent) who have applied the same modifications to enjoy a private shared space using existing rules without creating a DNA from scratch.

Author: you also made the sentence work way better.

#### Immune system

A property of Holochain's [validating DHT](#validating-dht), whereby healthy [nodes](#node) detect invalid data, share proof of corruption among their peers via [warrants](#warrant), and take defensive action against the corrupt nodes that [authored](#author) the data by blocking network communication with them. While each agent is individually free to interact with a warranted peer, most agents will refuse to interact or gossip with them. The cumulative effect is a collective exclusion of the corrupt nodes (see [mutual sovereignty](#mutual-sovereignty)).

#### Init callback

> A function in a [DNA](#dna) that the [conductor](#conductor) calls when an [agent](#agent) starts a [cell](#cell) for the first time, and after they have joined the DNA's [network](#network). This can be used to set up initial [source chain](#source-chain) [#state](#state-transition), etc.

Reviewer: ```suggestion
A function in a [DNA](#dna) that the [conductor](#conductor) executes when an [agent](#agent) calls a [cell](#cell) for the first time, and after they have joined the DNA's [network](#network). This can be used to set up initial [source chain](#source-chain) [#state](#state-transition), etc.
```

Author: Ah yes, this is true. I'd elided the fact that it didn't run until first zome call, but that's probably important lifecycle timing to know about.

#### Init complete action

> A [record](#record) that Holochain automatically writes to an [agent's](#agent) [source chain](#source-chain) to indicate that the [init callbacks](#init-callback) in all of a [DNA's](#dna) [zomes](#zome) have successfully run and the [cell](#cell) is ready to use.

Reviewer: Strictly the cell can be used before having been lazily initialized as part of the first zome call.
```suggestion
An [action](#action) that Holochain automatically writes to an [agent's](#agent) [source chain](#source-chain) to indicate that the [init callbacks](#init-callback) in all of a [DNA's](#dna) [zomes](#zome) have successfully run..
```

> A [record](#record) that Holochain automatically writes to an [agent's](#agent) [source chain](#source-chain) to indicate that the [init callbacks](#init-callback) in all of a [DNA's](#dna) [zomes](#zome) have successfully run and the [cell](#cell) is ready to use.

Author: ```suggestion
An [action](#action) that Holochain automatically writes to an [agent's](#agent) [source chain](#source-chain) to indicate that the [init callbacks](#init-callback) in all of a [DNA's](#dna) [zomes](#zome) have successfully run.
```

There was an extra period at the end of the suggestion; otherwise I would've accepted it ;)

#### Integrity zome

> A [zome](#zome) that defines a data schema. It does this through three specially named [zome functions](#zome-function):

Reviewer: Entry types and link types merely are enums in Rust. I was about to add `domain logic zome functions` to the coordinator zome above, but then thought that zome functions are exclusive to coordinator zomes and don't need that more specific descriptor. I wouldn't consider the validation function in integrity zomes a "zome function".
```suggestion
A [zome](#zome) that defines a data schema. It does this through three specially named constructs:
```

Author: oh, I thought the conductor retrieved them at installation time by calling an `entry_types` function?

#### Link types callback

A private [zome function](#zome-function) in an [integrity zome](#integrity-zome) that yields all the link types defined in the zome's schema. This callback is called by the [conductor](#conductor) at [DNA](#dna) installation time.

#### Live data

[DHT data](#dht-data) or [source chain](#source-chain) data that meets two criteria:

* It's been [validated](#validation-rule) and found valid.
* Its [CRUD](#crud-action) status doesn't mark it as [dead](#dead-data).

#### Lobby

A Holochain application design pattern, in which one [DHT](#distributed-hash-table-dht) is established as a common space which [agents](#agent) can join and either request access to a privileged DHT or ask privileged agents to mediate access to that DHT using [remote calls](#remote-call).

#### Logical monotonicity

> The property of a set of facts whereby the truth of prior facts are never negated by the addition of later facts. [CALM](#consistency-as-logical-monotonicity-calm-theorem) relies on functions that exhibit this property. For example, Holochain's [source chain](#source-chain), [DHT](#distributed-hash-table-dht) only add new data without removing old data, simulating deletions and modifications ([CRUD](#create-read-update-delete-crud)) by recording actions which override the status of, but don't remove, the data they refer to.

Author: ```suggestion
The property of a set of facts whereby the truth of prior facts are never negated by the addition of later facts. [CALM](#consistency-as-logical-monotonicity-calm-theorem) relies on functions that exhibit this property. For example, Holochain's [source chain](#source-chain) and [DHT](#distributed-hash-table-dht) only add new data without removing old data, simulating deletions and modifications ([CRUD](#create-read-update-delete-crud)) by recording actions which override the status of, but don't remove, the data they refer to.
```

#### Nearness

The mathematical distance between two [addresses](#address) to each other in the [DHT's](#distributed-hash-table-dht) [address space](#address-space).

#### Neighbor

See [neighborhood](#neighborhood).

#### Neighborhood

A range of [DHT addresses](#dht-address) about which a [node](#node) attempts to everything they ought to know. Neighbors collectively support the [resilience](#resilience) of all [DHT data](#dht-data) whose [address](#dht-address) is within their respective [storage arcs](#storage-arc) by storing and [validating](#validation-rule) it and [gossiping](#gossip) it to all [neighbors](#neighbor) with whom their storage arcs overlap.

<!--- removed because query arcs don't exist yet -- They also have a wider neighbourhood of nodes they can talk to to receive authoritative data, defined by their [query arc](#query-arc). -->

#### Network

In Holochain terms, a collection of [nodes](#node) [gossiping](#gossip) with each other to form a [validating DHT](#validating-dht), aiding in data storage and retrieval, [validation](#validation-rule), and peer discovery. Each [DNA](#dna) has a separate network.

#### Network seed

> An optional string, specified in a [DNA bundle](#dna-bundle) file or passed at DNA installation time, that modifies the [DNA's hash](#dna-hash) without modifying any of its behavior. This can be used to create a unique [network](#network) shared by all [agents](#agent) who use the same network seed. Hence, a network seed is considered a [DNA modifier](#dna-modifiers).

Author: ```suggestion
An optional string, specified in a [DNA bundle](#dna-bundle) file or passed at [cell](#cell) [cloning](#cloning) time, that modifies the [DNA's hash](#dna-hash) without modifying any of its behavior. This can be used to create a unique [network](#network) shared by all [agents](#agent) who use the same network seed. Hence, a network seed is considered a [DNA modifier](#dna-modifiers).
```

#### Sharding

A process of reducing the processing and storage load of individual [nodes](#node) in a [distributed system](#distributed-system) by distributing data and/or work among them. While some sharded systems such as [Ethereum 2](https://ethereum.org/en/eth2/) separate nodes into discrete shards, Holochain's [DHT](#distributed-hash-table-dht) separates them into [neighborhoods](#neighborhood) of overlapping [storage arcs](#storage-arc). Each node in a DHT takes responsibility to store a shard of the total public data in the DHT, according to the chosen size of their own storage arc.

#### Signal

1. A message emitted by a [cell](#cell), meant for a [client](#client) to receive and act upon.
2. A message sent by one cell to another in the same [network](#network).

#### Source chain

A [hash chain](#hash-chain) of [records](#record) committed by an [agent](#agent). Every agent has a separate source chain for each of the [cells](#cell) they're running, which stores all of the [actions](#action) or [state transitions](#state-transition) the cell has made.

#### State transition

> A modification of application state. In Holochain, all state transitions are initially created as [records](#record) in an [agent's](#agent) [source chain](#source-chain) that represent the [actions](#action) of [creating, updating, and deleting](#create-read-update-delete-crud) data and metadata, as well as of system-level actions such as [capability grants](#capability-grant). A state transition then yields one or more [operations](#dht-operation) that are then [published](#publish) to the [DHT](#distributed-hash-table-dht), that is, they are sent to the appropriate [validation authorities](#validation-authority), who then apply those operations to their own DHT [shard](#sharding), which causes a state transition for the [base](#base) to which the operation applies.

Reviewer: ```suggestion
A modification of application state. In Holochain, all state transitions are initially created as [records](#record) in an [agent's](#agent) [source chain](#source-chain) that represent the [actions](#action) of [creating, updating, and deleting](#create-read-update-delete-crud) data and metadata, as well as of system-level actions such as [capability grants](#capability-grant). A state transition further yields one or more [operations](#dht-operation) that are then [published](#publish) to the [DHT](#distributed-hash-table-dht), that is, they are sent to the appropriate [validation authorities](#validation-authority), who then apply those operations to their own DHT [shard](#sharding), which causes a state transition for the [base](#base) to which the operation applies.
```
